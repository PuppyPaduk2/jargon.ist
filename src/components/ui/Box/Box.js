// @flow

import React, { type Node } from 'react';
import cx from 'classnames';

import './Box.scss';

type Props = {
  children: Node,
  className?: string,
};

const Box = ({ children, className }: Props) => (
  <div className={cx('c-box', className)}>{children}</div>
);

Box.defaultProps = {
  className: '',
};

export default Box;
