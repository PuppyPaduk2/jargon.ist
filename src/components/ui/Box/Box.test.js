// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Box from './Box';

it('renders correctly', () => {
  const wrapper = shallow(
    <Box>
      <div>Example content</div>
    </Box>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('passes classname to wrapper', () => {
  const wrapper = shallow(
    <Box className="example-classname">
      <div>Example content</div>
    </Box>,
  );

  expect(wrapper.find('.example-classname').exists()).toEqual(true);
});
