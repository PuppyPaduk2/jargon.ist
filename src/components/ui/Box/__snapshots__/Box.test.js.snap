// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="c-box"
>
  <div>
    Example content
  </div>
</div>
`;
